### ISSUE TITLE
Update the directory of generated DLL files

### ISSUE THREAD

**Issue by liruiqi97** [OpenFAST/openfast#1325]
**Bug description**
I installed OpenFAST on my MacBook with `cmake` commands. The installation was successful, but some issues happened with the Regression Test. 
When running the [regression tests](https://github.com/OpenFAST/r-test.git), almost all those `5MW` tests were failed on my computer. The reason of the failure was that the project could not find `DISCON.dll`, `DISCON_ITIBarge.dll`, `DISCON_OC3Hywind.dll` in correct folders. Those DLL files were generated somewhere in the `openfast/build` folder. After I moved those DLL files properly, those tests became passed.

The original location of those files were
`openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON/build/DISCON.dll`
`openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON_ITI/build/DISCON_ITIBarge.dll`
`openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON_OC3/build/DISCON_OC3Hywind.dll`
`openfast/build/reg_tests/glue-codes/python/5MW_Baseline/ServoData/DISCON/build/DISCON.dll`
`openfast/build/reg_tests/glue-codes/python/5MW_Baseline/ServoData/DISCON_ITI/build/DISCON_ITIBarge.dll`
`openfast/build/reg_tests/glue-codes/python/5MW_Baseline/ServoData/DISCON_OC3/build/DISCON_OC3Hywind.dll`.
I moved those DLL files respectively into the following locations, and then the problem was solved.
`openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON.dll`
`openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON_ITIBarge.dll`
`openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON_OC3Hywind.dll`
`openfast/build/reg_tests/glue-codes/python/5MW_Baseline/ServoData/DISCON.dll`
`openfast/build/reg_tests/glue-codes/python/5MW_Baseline/ServoData/DISCON_ITIBarge.dll`
`openfast/build/reg_tests/glue-codes/python/5MW_Baseline/ServoData/DISCON_OC3Hywind.dll`.

Since I solved the problem by simply moving the files, there are probably some mistakes of file linking, in the settings of the regression tests.

**To Reproduce**
1. Download [OpenFAST v3.3.0](https://github.com/OpenFAST/openfast/releases/tag/v3.3.0), which folder will be called `openfast` in the following.
    Download the [unit test folder](https://github.com/Goddard-Fortran-Ecosystem/pFUnit.git) and use it to replace `openfast/unit_tests/pfunit`. 
    Download the [regression test folder](https://github.com/OpenFAST/r-test.git) and use it to replace `openfast/reg_tests/r-test`.
2. Install OpenFAST through the following commands on Terminal in MacBook. (The settings may differ for other users.)
    `cd openfast/build`
    `cmake .. -D CMAKE_Fortran_COMPILER=/usr/local/bin/gfortran-12`
    `cmake .. -D CMAKE_INSTALL_PREFIX=/usr/local/`
    `cmake .. -D BUILD_TESTING=ON`
    `make install`
3. Run the regression tests. Here we take a single test `5MW_Land_DLL_WTurb` as an example.
    `ctest -R 5MW_Land_DLL_WTurb`
4. Then, my observation was a failed result.
<img width="593" alt="Screen Shot 2022-11-15 at 1 26 31 PM" src="https://user-images.githubusercontent.com/43281955/201997330-ae7d8275-e46d-47bc-a8ed-1027dbba030b.png">
The log `openfast/build/Testing/Temporary/LastTest.log` showed that
<img width="630" alt="Screen Shot 2022-11-15 at 1 28 04 PM" src="https://user-images.githubusercontent.com/43281955/201997747-99e005f4-408c-47fb-b244-b835a1917433.png">

5. Open `openfast/build/reg_tests/glue-codes/openfast/5MW_Land_DLL_WTurb/5MW_Land_DLL_WTurb.log`, and this is what i saw at the end of the file.
<img width="597" alt="Screen Shot 2022-11-15 at 1 24 23 PM" src="https://user-images.githubusercontent.com/43281955/201997030-a777fe19-e8ec-4cec-802c-0c209cd91860.png">
It says that `DISCON.dll` was not found.

6. The issue was solved after i moved 
    `openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON/build/DISCON.dll`
to 
    `openfast/build/reg_tests/glue-codes/openfast/5MW_Baseline/ServoData/DISCON.dll`,
as I said on top.

**Expected behavior**
The regression test should be passed without moving any files by users.

**OpenFAST Version**
OpenFAST v3.3.0

```
**************************************************************************************************
 OpenFAST

 Copyright (C)  National Renewable Energy Laboratory
 Copyright (C)  Envision Energy USA LTD

 This program is licensed under Apache License Version 2.0 and comes with ABSOLUTELY NO WARRANTY.
 See the "LICENSE" file distributed with this software for details.
 **************************************************************************************************

 OpenFAST-v3.3.0
 Compile Info:
  - Architecture: 64 bit
  - Precision: double
  - Date: Nov 15 2022
  - Time: 13:20:00
 Execution Info:
  - Date: 11/15/2018
  - Time: 13:22:00
```

**System Information (please complete the following information):**
 - OS: macOS 12.4
 - Compiler: GFortran 12
 - Compiler settings:

**Comment by andrew-platt** [comment]
I wonder if this is in part caused by the setting the install location differently with `CMAKE_INSTALL_PREFIX=/usr/local/`.  We may have an issue with regression testing in this case.

Also, the `5MW_Land_DLL_WTurb` should not be giving a segmentation fault when the DLL is not found.  There may be a bug here as well that we don't know about.

I'll do some testing before our 3.4.0 release.

**Comment by liruiqi97** [comment]
Thanks Andrew for responding this!

**Comment by deslaughter** [comment]
@liruiqi97, thank you for the detailed issue description. I was able to reproduce the error about the missing DLL. This problem is due to the way that CMake sets up the files for the regression tests. To have the build system put the DLLs in the proper locations, please run `make regression_tests` after `make install`. I don't think this step is documented, but it was mentioned in OpenFAST/r-test#34.

@andrew-platt the `all` target does not depend on the `regression_tests` target. This can be fixed by adding `ALL` to line 184 fix in the `r-test/CMakeLists.txt` file. I suggest we go ahead and make the change in that repo.

**Comment by rafmudaf** [comment]
@deslaughter when I run only `make regression_tests`, I get the following output:

```
(base) >>mbp@~/Development/openfast/build (dev $)$ make regression_tests
Scanning dependencies of target DISCON_OC3Hywind
[  1%] Building Fortran object /Users/rmudafor/Development/openfast/reg_tests/r-test/glue-codes/openfast/5MW_Baseline/ServoData/DISCON_OC3/build/CMakeFiles/DISCON_OC3Hywind.dir/DISCON_OC3Hywind.F90.o
[  1%] Linking Fortran shared library DISCON_OC3Hywind.dll
[  1%] Built target DISCON_OC3Hywind
Scanning dependencies of target DISCON
[  1%] Building Fortran object /Users/rmudafor/Development/openfast/reg_tests/r-test/glue-codes/openfast/5MW_Baseline/ServoData/DISCON/build/CMakeFiles/DISCON.dir/DISCON.F90.o
[  1%] Linking Fortran shared library DISCON.dll
[  1%] Built target DISCON
Scanning dependencies of target DISCON_ITIBarge
[  1%] Building Fortran object /Users/rmudafor/Development/openfast/reg_tests/r-test/glue-codes/openfast/5MW_Baseline/ServoData/DISCON_ITI/build/CMakeFiles/DISCON_ITIBarge.dir/DISCON_ITIBarge.F90.o
[  1%] Linking Fortran shared library DISCON_ITIBarge.dll
[  1%] Built target DISCON_ITIBarge
[  1%] Generating glue-codes/openfast/5MW_Baseline/ServoData/DISCON.dll
[  1%] Generating glue-codes/openfast/5MW_Baseline/ServoData/DISCON_ITIBarge.dll
[  3%] Generating glue-codes/openfast/5MW_Baseline/ServoData/DISCON_OC3Hywind.dll
[  3%] Generating glue-codes/python/5MW_Baseline/ServoData/DISCON.dll
[  3%] Generating glue-codes/python/5MW_Baseline/ServoData/DISCON_ITIBarge.dll
[  3%] Generating glue-codes/python/5MW_Baseline/ServoData/DISCON_OC3Hywind.dll
[  3%] Generating glue-codes/fast-farm/5MW_Baseline/ServoData/DISCON_WT1.dll
[  4%] Generating glue-codes/fast-farm/5MW_Baseline/ServoData/DISCON_WT2.dll
[  4%] Generating glue-codes/fast-farm/5MW_Baseline/ServoData/DISCON_WT3.dll
```

Does this not happen for you?

**Comment by rafmudaf** [comment]
FWIW I'm not opposed to adding the regression test CMake targets to `ALL` as in #1376, but I'm surprised that the current configuration does not work.

**Comment by rafmudaf** [comment]
Ah my apologies, I now see that the issue in the original post is that the install prefix was changed to install to `/usr/local`. In that case, `make install` is not required for the reg tests, but `make regression_tests` should work.

**Comment by andrew-platt** [comment]
This should now be fixed with #1376
